src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGANaivelySpark.java

```diff
@@ -0,0 +1,485 @@
+import scala.Tuple3;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+// TODO: choose which parameters allowed to be tunable
+// TODO: if throws, would temp files be cleaned up automatically?
+// TODO: most robust way to check OS in method SGASerialRunner (Mac doesn't support multi-thread mechanism in SGA)?
+@CommandLineProgramProperties(
+        summary        = "Program to call SGA for performing local assembly and outputs assembled contigs.",
+        oneLineSummary = "Perform SGA-based local assembly on fasta files on Spark",
+        programGroup   = StructuralVariationSparkProgramGroup.class)
+public final class RunSGANaivelySpark extends GATKSparkTool {
+    private static final long serialVersionUID = 1L;
+
+    @Argument(doc       = "Absolute path to SGA installation.",
+              shortName = "sgaPath",
+              fullName  = "fullPathToSGA",
+              optional  = false)
+    public String pathToSGA = null;
+
+    @Argument(doc       = "A path to a directory to write results to.",
+              shortName = "outDir",
+              fullName  = "outputDirectory",
+              optional  = false)
+    public String outputDir = null;
+
+    @Argument(doc       = "Number of threads to use when running sga.",
+              shortName = "t",
+              fullName  = "threads",
+              optional  = true)
+    public int threads = 1;
+
+    @Argument(doc       = "To run k-mer based read correction, filter and duplication removal in SGA or not, with default parameters.",
+              shortName = "correct",
+              fullName  = "correctNFilter",
+              optional  = true)
+    public boolean runCorrectionSteps = false;
+
+    @Override
+    public boolean requiresReads(){
+        return true;
+    }
+
+    @Override
+    public void runTool(final JavaSparkContext ctx){
+
+        final JavaRDD<GATKRead> reads = getReads();
+        // TODO: experimental interface to caller, should be getting this from JavaSparkContext
+        final JavaPairRDD<Long, Iterable<GATKRead>> readsAroundBreakpoints = reads.groupBy( read -> 1L);
+
+        final Path sgaPath = Paths.get(pathToSGA);
+
+        // map paired reads to FASTA file (temp files live in temp dir that cleans self up automatically) then feed to SGA for assembly
+        final JavaPairRDD<Long, SGAAssemblyResultType> assembledContigs = readsAroundBreakpoints.mapToPair(RunSGANaivelySpark::convertReadsToFASTQ)
+                                                                                                .mapToPair(entry -> RunSGANaivelySpark.performAssembly(entry, sgaPath, threads, runCorrectionSteps));
+
+        validateAndSaveResults(assembledContigs, outputDir);
+    }
+
+    // validates the returned result:
+    //   if all steps return 0, save the contig file and may discard the runtime information
```

Comment:
This comment doesn't seem to quite describe what you are doing: you're saving the results of any steps that succeeded and the runtime info of any steps that failed.